h_malloc.c

```diff
@@ -1123,8 +1225,15 @@ COLD static void init_slow_path(void) {
     if (unlikely(memory_protect_rw_metadata(ra->regions, ra->total * sizeof(struct region_metadata)))) {
         fatal_error("failed to unprotect memory for regions table");
     }
-
+#ifdef HAS_ARM_MTE
+    if (likely(is_memtag_enabled())) {
+        ro.slab_region_start = memory_map_mte(slab_region_size);
+    } else {
+        ro.slab_region_start = memory_map(slab_region_size);
+    }
+#else
     ro.slab_region_start = memory_map(slab_region_size);
+#endif
```

Comment:
```suggestion
#ifdef HAS_ARM_MTE
    if (likely(is_memtag_enabled()))
        ro.slab_region_start = memory_map_mte(slab_region_size);
   else 
#else
    ro.slab_region_start = memory_map(slab_region_size);
#endif
```

Reply:
This change hurts readability I think, decided against it when writing these checks.

Reply:
Your call :)